website/content/docs/kubernetes/deploy.mdx

```diff
@@ -0,0 +1,177 @@
+---
+layout: docs
+page_title: Kubernetes - YAML-Free Deployment
+description: |-
+  For many applications, Waypoint can deploy to Kubernetes with only a handful of lines of configuration in the `waypoint.hcl` file and zero external configuration.
+---
+
+# Deploying Applications without YAML
+
+For many applications, Waypoint can deploy to Kubernetes with only
+a handful of lines of configuration in the `waypoint.hcl` file and zero
+external configuration. For new applications, this is a great option to
+get up and running easily. If your application becomes more complex later,
+you can always switch deployment plugins.
+
+-> **Prefer writing YAML? That's okay!** Waypoint supports
+[Helm](/docs/kubernetes/helm-deploy) and raw
+[`kubectl apply`](/docs/kubernetes/kubectl-deploy) as first-class
+deployment plugins, too.
+
+## A Simple Application
+
+For a simple web application, you can get started with almost zero configuration:
+
+```hcl
+app "my-app" {
+  deploy {
+    use "kubernetes" {}
+  }
+}
+```
+
+With zero configuration, Waypoint will create a Kubernetes `Deployment`
+with a `PORT` environment variable that your application is expected to listen
+on. And Waypoint will expose this application using a `ClusterIP` service.
+This is a perfect, minimal deployment for an internal service.
+
+-> **Common mistake:** your application must be written to listen on the
+`PORT` environment variable for this to _just work_. If you want to change
+the port to something else, specify it using the `ports` configuration
+shown below.
+
+### Scaling Up
+
+With one more line of configuration, you can scale your application to more repliacs:
+
+```hcl
+app "my-app" {
+  deploy {
+    use "kubernetes" {
+      replicas = 3
+    }
+  }
+}
+```
+
+### Health Checking
+
+With one line of configuration, you can specify an HTTP health check that
+Kubernetes will use for both liveness and readiness. This HTTP endpoint
+should listen on the service port and respond with a `2XX` status code
+for success.
+
+```hcl
+app "my-app" {
+  deploy {
+    use "kubernetes" {
+      probe_path = "/_healthz"
+    }
+  }
+}
+```
+
+### Alternate or Additional Ports
+
+With zero configuration, your application must listen on the `PORT`
+environment variable set by Waypoint. You may change or add ports
+using the `ports` variable:
+
+```hcl
+app "my-app" {
+  deploy {
+    use "kubernetes" {
+      ports = [{
+        port = 3000
+        name = "http"
+      }]
+    }
+  }
+}
+```
+
+-> **Important! The first port is special.** The first port is used
+as the expected HTTP listening port (regardless of name or any other
+settings).
+
+### Autoscaling
+
+Waypoint can configure [horizontal autoscaling](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/)
+in just a few lines:
+
+```hcl
+app "my-app" {
+  deploy {
+    use "kubernetes" {
+      autoscale {
```

Comment:
Metrics server also needs a CPU resource limit/request defined otherwise k8s won't scale out. So we should include the `cpu` stanza here for a complete picture